### PR TITLE
Bugfix for pledge_sent_* updating at inappropriate times -- 1286 [WIP]

### PR DESCRIPTION
**Pull request by mdworken**
I rule and have completed some work on Case Manager that's ready for review!

Fixes issues in #1286 

This pull request makes the following changes:
Adds logic in controller to prevent pledge_sent_by and pledge_sent_at from updating when they shouldn't.

(If there are changes to the views, please include a screenshot so we know what to look for!)

It relates to the following issue #s:

### CODE REVIEW

File: app/controllers/patients_controller.rb [hunk 1]
```diff
@@ -58,9 +58,13 @@ def edit
   end
 
   def update
-    if @patient.update_attributes params[:pledge_sent]
-       @patient.pledge_sent_at = Time.zone.now
-       @patient.pledge_sent_by = current_user
+    @patient.update_attributes params[:pledge_sent]
```
Comment (colinxfleming) [inline app/controllers/patients_controller.rb]: I think this additional logic fixes the bug at hand, nicely done!

A quick rails-ism: `@patient` is stored independently of the patient in memory, so after running the update statement (which returns `true` or `false` based on whether the update transaction completes or not), we'll have a disagreement between the patient object we're working with and the actual representation in db. We can resolve this disagreement with a `@patient.reload`.

tldr: after this `update_attributes` , throw in `@patient.reload` and then this should be good to merge. (This should also fix your test failure, if I'm reading it right.)

Let me know if that makes sense! Thanks for knocking this out, really appreciate it.

Comment (mdworken) [reply app/controllers/patients_controller.rb]: Great, thanks for the help! I'm headed to a holiday party right now, I'll fix it early tomorrow morning :)
